modules/kafka/kafka.go

```diff
@@ -72,12 +72,17 @@ func RunContainer(ctx context.Context, opts ...testcontainers.ContainerCustomize
 							return err
 						}
 
+						ip, err := c.ContainerIP(ctx)
```

Comment:
[this message](https://github.com/testcontainers/testcontainers-go/pull/1786/files#r1367487506) is still valid in this PR and align implementations. It will make things easier to use the hostname rather the ip when getting access using a container network.

Reply:
I tested your PR with the hostname and in my case it broke running the tests via `go test`. I'm not sure if there is some (easy) way for the host to resolve containers via hostname

Reply:
about the error in the description

> Currently it leads to errors in the TransactionCoordinator:
WARN [TransactionCoordinator id=1] Connection to node 1 (/172.17.0.1:9092) could not be established. Broker may not be available. (org.apache.kafka.clients.NetworkClient)

The client should connect to the mapped port in 9093 instead. I think I am missing something in your setup. Can you elaborate more, please?

Reply:
The error is thrown from inside the kafka container. The TransactionCoordinator is a Kafka service running on the broker and it tries to reach the broker it is running on. It for some reason can't resolve the container name as hostname (this I also find a bit strange, because it should run in the docker network)

Reply:
nice, that works out :)